pcsx2/DEV9/Win32/tap-win32.cpp

```diff
@@ -298,9 +301,18 @@ bool TAPAdapter::recv(NetPacket* pkt)
 		DWORD dwError = GetLastError();
 		if (dwError == ERROR_IO_PENDING)
 		{
-			WaitForSingleObject(read.hEvent, INFINITE);
-			result = GetOverlappedResult(htap, &read,
-										 &read_size, FALSE);
+			HANDLE readHandles[]{read.hEvent, cancel};
+			const DWORD waitResult = WaitForMultipleObjects(2, readHandles, FALSE, INFINITE);
+
+			if (waitResult == WAIT_OBJECT_0 + 1)
+			{
+				CancelIo(htap);
+				//Wait for the I/O subsystem to acknowledge our cancellation
+				result = GetOverlappedResult(htap, &read, &read_size, TRUE);
+			}
+			else
+				result = GetOverlappedResult(htap, &read, &read_size, FALSE);
+
 			if (!result)
 			{
 			}
```

Comment:
Is there much point in an empty if statement?

Reply:
I guess not, I assume the idea was to have some kind-of error handling, but that if statement has been left that way since the plugin was added to PCSX2's repo many years ago.

May aswell just remove it this point.